### ISSUE TITLE
Setting environment variables results in role defined in credentials file being ignored

### ISSUE THREAD

**Issue by ysgard** [hashicorp/aws-sdk-go-base#13]
Might be related to: https://github.com/hashicorp/aws-sdk-go-base/issues/4

We're having an odd issue where we set a role (along with our keys) in ~/.aws/credentials, like so:

```
[default]
aws_access_key_id = <redacted>
aws_secret_access_key = <redacted>

[demo]
role_arn = arn:aws:iam::<redacted>:role/OrganizationAccountAccessRole
source_profile = default
```

And this works just fine with the provider:
```
provider "aws" {
  alias   = "demo"
  region  = "us-west-2"
  profile = "demo"
}
```

But when the credentials are set in the environment, terraform fails to work.

```
$ export AWS_ACCESS_KEY_ID=<redacted>
$ export AWS_SECRET_ACCESS_KEY=<redacted>
$ terraform plan
...
Error: error reading <resources defined in the demo account>

$ unset AWS_ACCESS_KEY_ID
$ unset AWS_SECRET_ACCESS_KEY
$ terraform plan
...
<no errors>
```

This is with
```
$ terraform version
Terraform v0.12.8
+ provider.aws v2.24.0
```

**Comment by mhumeSF** [comment]
Similar setup where on an ec2 instance I have just a `~/.aws/config`
```
[profile demo]
role_arn=arn:aws:iam::123456789000:role/TerraformApply
credential_source=Ec2InstanceMetadata
```

main.tf
```
provider "aws" {
  region = "us-west-2"
}

terraform {
  backend "s3" {
    bucket                  = "terraform-state-bucket" # s3 bucket in demo account
    key                     = "foo/terraform.tfstate"
    region                  = "us-west-2"
  }
}

data "aws_caller_identity" "current" {}

output "account_id" {
  value = "${data.aws_caller_identity.current.account_id}"
}
```

I then `export AWS_PROFILE=demo` and aws commands work, but terraform init fails not picking up AWS_PROFILE.

**Comment by bflad** [comment]
Hi folks 👋 These issues are likely resolved by #20, which has been merged and will release with v0.5.0 of this library, for inclusion with Terraform CLI (S3 Backend) v0.13.0-rc1 and Terraform AWS Provider v3.0.0.

If you are still having trouble once those updated versions of the Terraform S3 Backend and/or AWS Provider have been released. Please create a new issue and we'll take a fresh look. 👍
